collect_app/src/test/java/org/odk/collect/android/formlists/blankformlist/BlankFormListViewModelTest.kt

```diff
@@ -0,0 +1,363 @@
+package org.odk.collect.android.formlists.blankformlist
+
+import android.app.Application
+import androidx.lifecycle.MutableLiveData
+import androidx.test.core.app.ApplicationProvider
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import junit.framework.TestCase.assertFalse
+import org.hamcrest.CoreMatchers.`is`
+import org.hamcrest.MatcherAssert.assertThat
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.mockito.kotlin.doReturn
+import org.mockito.kotlin.mock
+import org.mockito.kotlin.verify
+import org.mockito.kotlin.verifyNoInteractions
+import org.mockito.kotlin.whenever
+import org.odk.collect.analytics.Analytics
+import org.odk.collect.android.external.FormsContract
+import org.odk.collect.android.formmanagement.FormsUpdater
+import org.odk.collect.android.formmanagement.matchexactly.SyncStatusAppState
+import org.odk.collect.android.preferences.utilities.FormUpdateMode
+import org.odk.collect.android.utilities.ChangeLockProvider
+import org.odk.collect.android.utilities.FormsDirDiskFormsSynchronizer
+import org.odk.collect.androidtest.LiveDataTester
+import org.odk.collect.forms.Form
+import org.odk.collect.forms.FormSourceException
+import org.odk.collect.formstest.FormUtils
+import org.odk.collect.formstest.InMemFormsRepository
+import org.odk.collect.settings.keys.ProjectKeys
+import org.odk.collect.shared.settings.InMemSettings
+import org.odk.collect.testshared.BooleanChangeLock
+import org.odk.collect.testshared.TestScheduler
+
+@RunWith(AndroidJUnit4::class)
+class BlankFormListViewModelTest {
+    private val form1 = Form.Builder()
```

Comment:
I've improved it a bit. Maybe not entirely in a way you described because I wanted to avoid converting Form to kotlin but it should be a bit clearer now.